backend/src/main/java/meetup/connect/meetupevent/GoogleCalendar.java

```diff
@@ -0,0 +1,204 @@
+import org.springframework.util.CollectionUtils;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class GoogleCalendar {
+    private static final String APPLICATION_NAME = "MeetUpConnect Calendar";
+
+    /** Global instance of the JSON factory. */
+    private static final JsonFactory JSON_FACTORY = GsonFactory.getDefaultInstance();
+
+    /** Directory to store authorization tokens for this application. */
+    private static final String TOKENS_DIRECTORY_PATH = "token";
+
+    private static final String CALENDAR_ID = "primary";
+    private static final Logger log = LoggerFactory.getLogger(GoogleCalendar.class);
+
+    /**
+     * Global instance of the scopes required by this quickstart. If modifying these scopes, delete
+     * your previously saved tokens/ folder.
+     */
+    private static final List<String> SCOPES = List.of(CalendarScopes.CALENDAR_EVENTS);
+
+    private static final String CREDENTIALS_FILE_PATH = "/credentials.json";
+    private static Calendar service;
+
+    /**
+     * Creates an authorized Credential object.
+     *
+     * @param HTTP_TRANSPORT The network HTTP Transport.
+     * @return An authorized Credential object.
+     * @throws IOException If the credentials.json file cannot be found.
+     */
+    private static Credential getCredentials(final NetHttpTransport HTTP_TRANSPORT)
+            throws IOException {
+        // Load client secrets.
+        InputStream in = GoogleCalendar.class.getResourceAsStream(CREDENTIALS_FILE_PATH);
+        if (in == null) {
+            throw new FileNotFoundException("Resource not found: " + CREDENTIALS_FILE_PATH);
+        }
+        GoogleClientSecrets clientSecrets =
+                GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+
+        // Build flow and trigger user authorization request.
+        GoogleAuthorizationCodeFlow flow =
+                new GoogleAuthorizationCodeFlow.Builder(HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES)
+                        .setDataStoreFactory(new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+                        .setAccessType("offline")
+                        .build();
+        LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(8888).build();
+        // returns an authorized Credential object.
+        return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+    }
+
+    private static Calendar getService() throws GeneralSecurityException, IOException {
+        final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+        return new Calendar.Builder(HTTP_TRANSPORT, JSON_FACTORY, getCredentials(HTTP_TRANSPORT))
```

Comment:
suggestion:
```suggestion
                final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
        var requestFactory = HTTP_TRANSPORT
                .createRequestFactory(httpRequest -> httpRequest
                        .setConnectTimeout(10)
                        .setReadTimeout(5)
                        .setNumberOfRetries(3)
                )
                .getTransport();
        return new Calendar.Builder(requestFactory, JSON_FACTORY, getCredentials(HTTP_TRANSPORT))
                .setApplicationName(APPLICATION_NAME)
                .build();
```

Reply:
let's adjust those timeouts

Reply:
Timeouts by default are set to 20 seconds https://cloud.google.com/java/docs/reference/google-http-client/latest/com.google.api.client.http.HttpRequest#com_google_api_client_http_HttpRequest_setConnectTimeout_int_, it's a little bit too long IMHO